### PR TITLE
[tensorflow] merge upstream sourcekit vfs changes

### PR DESCRIPTION
**Pull request by marcrasi**
Our SourceKit VFS changes have been upstreamed in https://github.com/apple/swift/pull/26168! This will cause conflicts with the SourceKit VFS changes in the `tensorflow` branch, so this PR preemptively merges those change into `tensorflow`.

This PR is organized into 3 commits:
* Revert all of the `tensorflow` SourceKit VFS changes.
* All of https://github.com/apple/swift/pull/26168, squashed.
* Shims for LangSupport functions that we are calling from our google-internal client. These shims will let us import these changes into google without modifying the client. After these changes are imported, I will update the client to use the new methods, and then delete the shims.

### CODE REVIEW

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

File: tools/SourceKit/include/SourceKit/Core/LangSupport.h [hunk 1]
```diff
@@ -730,9 +741,8 @@ class LangSupport {
   virtual void
   getCursorInfo(StringRef Filename, unsigned Offset, unsigned Length,
                 bool Actionables, bool CancelOnSubsequentRequest,
-                ArrayRef<const char *> Args,
-                llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem,
-                std::function<void(const RequestResult<SourceKit::CursorInfoData> &)> Receiver) = 0;
+                ArrayRef<const char *> Args, Optional<VFSOptions> vfsOptions,
+       std::function<void(const RequestResult<CursorInfoData> &)> Receiver) = 0;
```
Comment (burmako) [inline tools/SourceKit/include/SourceKit/Core/LangSupport.h]: Formatting looks a bit weird here.

File: tools/SourceKit/include/SourceKit/Support/FileSystemProvider.h [hunk 2]
```diff
@@ -31,27 +31,30 @@ namespace SourceKit {
...
+/// - source.request.cursorinfo:
+///     Uses the given custom filesystem. If none is specified, uses the
+///     filesystem associated with the file.
+/// - source.request.codecomplete:
+///     Uses the given custom filesystem.
+/// - source.request.codecomplete.open:
+///     Associates the given custom filesystem with this completion session, so
+///     that all subsequent requests in the session use it.
+/// - source.request.codecomplete.update:
+///     Uses the filesystem associated with the completion session.
```
Comment (burmako) [inline tools/SourceKit/include/SourceKit/Support/FileSystemProvider.h]: This is a lot of work, and I'm super happy that it made it into `master`!

**Review by burmako — APPROVED** [review]

File: tools/SourceKit/include/SourceKit/Core/LangSupport.h (hunk 1 above)
Comment (marcrasi) [reply tools/SourceKit/include/SourceKit/Core/LangSupport.h]: I think this is an intentional thing that is done some places in the Swift codebase to keep lines under the column limit. Anyways, this is from upstream so I won't make any changes.

**Pull request merged**